docs/sweep_attacks.md

```diff
@@ -0,0 +1,87 @@
+# Sweep Attacks
+
+Armory supports running adversarial attacks which "sweep" over a range of values for specified 
+attack parameters (e.g. `"eps"`). This helps automate the process of determining at what 
+attack parameter values (i.e. perturbation budget) a defense is no longer robust.
+
+To enable such an attack, set `attack_config["type"]` to `"sweep"`. 
+```aidl
+"attack": {
+    "module": "art.attacks.evasion",
+    "name": "ProjectedGradientDescent",
+    ...
+    "type": "sweep"
+}
+```
+
+Next, specify which parameter(s) to perform the search over. This is done using the 
+`attack_config["sweep_params"]["kwargs"]` field for kwargs passed to attack instantiation and the
+`attack_config["sweep_params"]["generate_kwargs"]` field for kwargs passed to the attack's 
+`generate()` method. In the example below we sweep over the `"eps"` and `"eps_step"` 
+parameters:
+
+```aidl
+"attack": {
+    "module": "art.attacks.evasion",
+    "name": "ProjectedGradientDescent",
+    "type": "sweep",
+    "sweep_params": {
+        "kwargs": {
+            "eps": [0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08],
+            "eps_step": [0.005, 0.01, 0.015, 0.02, 0.025, 0.03, 0.035, 0.04]
+        }
+    }
+}
+```
+
+Similarly, in the following example, we sweep over kwargs passed to the attack's `generate()` method 
+using the `"generate_kwargs"` field inside `attack_config["sweep_params"]`:
+```aidl
+"attack": {
+    "module": "armory.art_experimental.attacks.pgd_patch",
+    "name": "PGDPatch",
+    "type": "sweep",
+    "sweep_params": {
+        "generate_kwargs": {
+            "patch_height": [10, 20, 30, 40, 50],
+            "patch_width": [10, 20, 30, 40, 50]
+        }
+    }
+}
+```
+
+Each range of sweep parameters must be specified with a list of length `N`, where `N > 1` is 
+the number of desired search points. All parameters specified inside of 
+`attack_config["sweep_params"]` must correspond to lists of length `N`, and the `ith` element of
+a given kwarg list will be used in conjunction with the `ith` element of all other kwarg 
+lists (i.e. in the first example, when `"eps"` is `0.01`, `"eps_step"` is `0.005` and so on). The
+search algorithm assumes that parameters lists are in ascending order of attack strength.
```

Comment:
For completeness, maybe we should mention in the docs that the attack returns the weakest strength adversarial example that is a successful attack based on the metrics, or the original input if the attack fails.

Reply:
added